### PR TITLE
added more modules to docs lib

### PR DESCRIPTION
**Pull request by AFaurholt**
Per [Issue #16656](https://github.com/nim-lang/Nim/issues/16656) I tried to add the missing modules to the list. I wasn't sure what to do with io or assertions since they're listed under [system](https://nim-lang.org/docs/system.html)

### CODE REVIEW

File: doc/lib.rst [hunk 1]
```diff
@@ -249,6 +249,9 @@ Generic Operating System Services
   (also called *console*). The implementation simply uses ANSI escape
   sequences and does not depend on any other module.
 
+* `rdstdin <rdstdin.html>`_
+  This module contains code for reading from stdin. On UNIX the linenoise library is wrapped and set up to provide default key bindings (e.g. you can navigate with the arrow keys). On Windows system.readLine is used. This suffices because Windows' console already provides the wanted functionality.
```
Comment (konsumlamm) [inline doc/lib.rst]:
```suggestion
  This module contains code for reading from stdin.
```
I think this is enough.

File: doc/lib.rst [hunk 2]
```diff
@@ -354,6 +357,12 @@ Parsers
 * `json <json.html>`_
   High-performance JSON parser.
 
+* `jsonutils <jsonutils.html>`_
+  This module implements a hookable (de)serialization for arbitrary types. Design goal: avoid importing modules where a custom serialization is needed; see strtabs.fromJsonHook,toJsonHook for an example.
```
Comment (konsumlamm) [inline doc/lib.rst]:
```suggestion
  This module implements a hookable (de)serialization for arbitrary types.
```

File: doc/lib.rst [hunk 3]
```diff
@@ -354,6 +357,12 @@ Parsers
...
+
+* `parsejson <parsejson.html>`_
+  This module implements a json parser. It is used and exported by the json standard library module, but can also be used in its own right.
```
Comment (konsumlamm) [inline doc/lib.rst]:
```suggestion
  This module implements a JSON parser. It is used and exported by the `json` standard library module, but can also be used in its own right.
```

File: doc/lib.rst [hunk 4]
```diff
@@ -471,6 +480,13 @@ Miscellaneous
 * `std/varints <varints.html>`_
   Decode variable-length integers that are compatible with SQLite.
 
+* `enumerate <enumerate.html>`_
+  This module implements enumerate syntactic sugar based on Nim's macro system.
```
Comment (konsumlamm) [inline doc/lib.rst]:
```suggestion
  This module implements `enumerate` syntactic sugar based on Nim's macro system.
```

File: doc/lib.rst [hunk 5]
```diff
@@ -471,6 +480,13 @@ Miscellaneous
...
+
+* `with <with.html>`_
+  This module implements the with macro for easy function chaining. See `<https://github.com/nim-lang/RFCs/issues/193>`_ and `<https://github.com/nim-lang/RFCs/issues/192>`_ for details leading to this particular design.
```
Comment (konsumlamm) [inline doc/lib.rst]:
```suggestion
  This module implements the `with` macro for easy function chaining.
```

**Review by konsumlamm — CHANGES_REQUESTED** [review]

File: doc/lib.rst (hunk 1 above)
Comment (konsumlamm) [reply doc/lib.rst]: Also, this should go under the `Impure libraries` section, just add a new sub-section there.

**Comment by konsumlamm** [comment]
The modules should be ordered alphabetically and the modules in the `std` directory (`jsonutils`, `enumerate`, `with`) should have a `std/` prefix (at least until #16999 is merged).

File: doc/lib.rst (hunk 3 above)
Comment (ghost) [reply doc/lib.rst]: isn't "standard library" redundant?

```suggestion
  This module implements a JSON parser. It is used and exported by the `json` module, but can also be used in its own right.
```

Comment (ghost) [reply doc/lib.rst]: or we can actually make it a link to the json module

Comment (konsumlamm) [reply doc/lib.rst]: You're right, good idea.

**Comment by AFaurholt** [comment]
Added the std/ and the other suggestions

**Comment by konsumlamm** [comment]
See https://github.com/nim-lang/Nim/pull/17430#discussion_r598141532.

**Comment by AFaurholt** [comment]
> 
> 
> See [#17430 (comment)](https://github.com/nim-lang/Nim/pull/17430#discussion_r598141532).

Should be fixed now. Thanks for the catch

**Pull request merged**